utilities/zookeeper/R/corrector-aardvark.R

```diff
@@ -158,37 +205,65 @@ aardvark_state_corrections_single_signal <- function(x, params) {
       flag = .data$flag & # no corrections after some date
         (.data$time_value < ymd(params$time_value_flag_date) |
            .data$value < -params$size_cut),
+      # RI is not included in the special flagging process because it's sort of regular correlation 
       flag_bad_RI = (.data$geo_value == "ri" &
                        .data$value > 0 &
-                       abs(lag(.data$value) < 1e-6)),
-      flag_bad_WA = (.data$geo_value == "wa" & .data$signal == "deaths_incidence_num") &
-        (.data$time_value %in% ymd(c("2020-12-16", "2020-12-17",
-                                     "2020-12-23", "2020-12-24",
-                                     "2020-12-29"))),
-      #flag_bad_NC = (state == "NC" & time_value > ymd("2020-10-22")),
-      flag_bad_OH = (.data$geo_value == "oh" & .data$signal == "deaths_incidence_num") &
-        (.data$time_value %in% seq(ymd("2021-02-12"),
-                                   ymd("2021-02-14"), length.out = 3)),
-      flag_bad_VA = (.data$geo_value == "va" & .data$signal == "deaths_incidence_num") &
-        (.data$time_value > ymd("2021-02-20")),
-      corrected = corrections_multinom_roll(
-        .data$value, .data$value, .data$flag_bad_RI, .data$time_value, 7),
-      corrected = corrections_multinom_roll(
-        .data$value, .data$value, .data$flag_bad_WA, .data$time_value, 14),
-      corrected = corrections_multinom_roll(
-        .data$value, .data$value, .data$flag_bad_OH, .data$time_value, 60),
-      corrected = corrections_multinom_roll(
-        .data$value, .data$value, .data$flag_bad_VA, .data$time_value, 60),
-      corrected = corrections_multinom_roll( # for everywhere else
-        .data$corrected, .data$value,
-        (.data$flag &
-           !.data$flag_bad_RI & !.data$flag_bad_WA &
-           !.data$flag_bad_OH & !.data$flag_bad_VA),
-        .data$time_value, params$backfill_lag,
-        reweight = function(x) exp_w(x, params$backfill_lag)),
-      corrected = .data$corrected + # imputes forward if necessary
-        missing_future(TRUE, .data$time_value, .data$value, .data$fmean)
-    )
+                       abs(lag(.data$value) < 1e-6))))
+ 
+  # Check is there is special correction to make
+  if (all(is.na(special_flags)) == TRUE){
+    x <- x 
+  }else{
+    x <- special_flagging(x, special_flags)
+  } 
+  
+  # Correction on Rhode Island 
+  x_temp <- x %>% mutate(corrected = corrections_multinom_roll(
+    .data$value, .data$value, .data$flag_bad_RI, .data$time_value, 7))
+
+  # Correction on states that require special correction
+  x_temp$corrected <- special_correction(x_temp,special_flags) 
```

Comment:
Ok. Let's make `special_correction()` return the original df with 2 additional columns
1. `corrected`
2. `special_flag`
Then the second can tell us whether or not to correct that spot again. We can then remove all the copying below. Finally, on line 258, you can pass `.data$flag_loc & !.data$special_flag`.